### PR TITLE
Hubot: Adds some missing, new, types to bring up to date for v3

### PR DESCRIPTION
**Pull request by jphenow**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/hubotio/hubot
- [X] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [X] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/hubot/index.d.ts [hunk 1]
```diff
@@ -1,17 +1,63 @@
...
 // Project: https://github.com/hubotio/hubot
 // Definitions by: Dirk Gadsden <https://github.com/dirk>
 //                 Kees C. Bakker <https://github.com/KeesCBakker>
 //                 Emil Marklund <https://github.com/eeemil>
+//                 Jon Phenow <https://github.com/jphenow>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.3
-
-/// <reference types="node" />
+// TypeScript Version: 3.7
```
Comment (amcasey) [inline types/hubot/index.d.ts]: Out of curiosity, which 3.7 features do these new types depend on?

File: types/hubot/index.d.ts [hunk 2]
```diff
@@ -1,17 +1,63 @@
...
 
 import { EventEmitter } from 'events';
+import { Options as HttpOptions, ScopedClient } from 'scoped-http-client';
+import { Server } from 'http';
+import { Express } from 'express';
 
 declare namespace Hubot {
-    class Brain<A> extends EventEmitter {
+    interface UnknownData {
```
Comment (amcasey) [inline types/hubot/index.d.ts]: This doesn't seem to provide a lot of value compared to just using `{}`.

**Review by amcasey — CHANGES_REQUESTED** [review]
Minor questions.

File: types/hubot/index.d.ts (hunk 1 above)
Comment (jphenow) [reply types/hubot/index.d.ts]: Oh when I started I didn't realize it had more meaningful implications. I set it to the version _I_ was on. I can drop it to the lowest common denominator.

File: types/hubot/index.d.ts (hunk 2 above)
Comment (jphenow) [reply types/hubot/index.d.ts]: Yea it seemed more interesting when I started it. I'll replace the usages.

File: types/hubot/index.d.ts (hunk 1 above)
Comment (jphenow) [reply types/hubot/index.d.ts]: Was able to drop to 3.0 - needed to go that high for `unknown`

**Review by amcasey — APPROVED** [review]
Thanks for the changes!

**Comment by jphenow** [comment]
🙌 thank _you_!

**Pull request merged**